tests/core/pyspec/eth2spec/test/phase0/sanity/test_blocks.py

```diff
@@ -1038,7 +1041,18 @@ def test_historical_batch(spec, state):
 
     assert state.slot == block.slot
     assert spec.get_current_epoch(state) % (spec.SLOTS_PER_HISTORICAL_ROOT // spec.SLOTS_PER_EPOCH) == 0
-    assert len(state.historical_roots) == pre_historical_roots_len + 1
+
+    # check history update
+    if is_post_capella(spec):
+        # Frozen `historical_roots`
+        assert state.historical_roots == pre_historical_roots
+        if spec.fork == EIP4844:
+            # TODO: no-op for now in EIP4844 testnet
+            assert state.historical_batches == pre_historical_batches
+        else:
+            assert len(state.historical_batches) == len(pre_historical_batches) + 1
```

Comment:
we could check that historical_roots didn't change here, though it would eventually be caught by a state root check..

Reply:
added 5 lines above!